yaml_files/cuba.yml

```diff
@@ -565,3 +555,59 @@ CUBA_KEYS:
     definition: Thermodynamic ensemble
     shape: [1]
     type: string
+
+  LATTICE_UC_ABC:
+    definition: Lattice translation vectors lengths (a, b, c) of an atomic unit cell
+    shape: [3]
+    type: double
+
+  LATTICE_UC_ANGLES:
+    definition: Alpha, Beta and Gamma angles of an atomic unit cell
+    shape: [3]
+    type: double
+
+  SYMMETRY_GROUP:
+    definition: Symmetry group number acording to the Volume A of International Tables for Crystallography
+    shape: [1]
+    type: integer
+
+  CRYSTAL_ORIENTATION_1:
+    definition: Crystallographic orientation defined by Miller Index of the reference surface
+    shape: [3, 3]
+    type: double
+
+  CRYSTAL_ORIENTATION_2:
+    definition: Crystallographic orientation defined by Miller Index of the non-reference surface
+    shape: [3, 3]
+    type: double
+
```

Comment:
Yes, you are right, in principle there is only one crystallographic orientation defined by Index Miller (h,k,l). However, this description leave 1 degree freedom (the rotation around the axe defined by the Index Miller direction). This is usually not important because the scientist usually works in theoretical framework, with a fixed reference system. However, When you translate this to the "real" 3D space, n in determination appears. This indeterminacy must be defined by user with another vector that we called Crystal orientation 2. This value can be understand as a fake index Miller of one of the perpendicular direction to the Index Miller. Of course, not all the values are permitted for this   fake index miller, because it is conditioned by the real Index Miller and the crystal structure. We check the input given by the user, if it is correct we used, if not we used the closest correct orientation vector. 
In nCAD GUI we also give the opportunity to give an angle to define this indeterminacy. The problem is that the angle is not completely defined because it is not defined a 0 for this 0 must to arbitrary chosen (even if you use a criterion you are setting it arbitrarily). In the GUI it is easier to use it because it is graphically guided, but in an API it is a little bit confusing.